userComponent/src/test/kotlin/com/gravatar/app/usercomponent/data/RealUserRepositoryTest.kt

```diff
@@ -0,0 +1,128 @@
+package com.gravatar.app.usercomponent.data
+
+import com.gravatar.app.testUtils.CoroutineTestRule
+import com.gravatar.restapi.models.Avatar
+import com.gravatar.restapi.models.Profile
+import com.gravatar.services.AvatarService
+import com.gravatar.services.ErrorType
+import com.gravatar.services.GravatarResult
+import com.gravatar.services.ProfileService
+import com.gravatar.types.Hash
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.net.URI
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class RealUserRepositoryTest {
+
+    private val testDispatcher = StandardTestDispatcher()
+
+    @get:Rule
+    var coroutineTestRule = CoroutineTestRule(testDispatcher)
+
+    private lateinit var repository: RealUserRepository
+    private val avatarService = mockk<AvatarService>()
+    private val profileService = mockk<ProfileService>()
+    private val tokenStorage = FakeAuthTokenStorage()
+
+    private val testToken = "test-token"
+    private val testHash = "test-hash"
+
+    @Before
+    fun setup() {
+        repository = RealUserRepository(
+            avatarService = avatarService,
+            profileService = profileService,
+            tokenStorage = tokenStorage
+        )
+    }
+
+    @Test
+    fun `getAvatars should return success when user is logged in and services return success`() =
+        runTest {
+            // Given
+            val profile = createTestProfile()
+            val avatars = createTestAvatars()
+
+            // Save token
+            tokenStorage.save(testToken)
+
+            // Mock profile service
+            val profileResult = GravatarResult.Success<Profile, ErrorType>(profile)
+            coEvery { profileService.retrieveAuthenticatedCatching(testToken) } returns profileResult
+
+            // Mock avatar service
+            val avatarResult = GravatarResult.Success<List<Avatar>, ErrorType>(avatars)
+            coEvery {
+                avatarService.retrieveCatching(testToken, any<Hash>())
```

Comment:
`Hash` doesn't have `.equals` nor `.hashCode` functions so the matchers don't work well here and I had to use `any`.

Reply:
This should do the trick:

```suggestion
                avatarService.retrieveCatching(testToken, match { it.toString() == testHash })
```